packages/claims/src/claims/claims.ts

```diff
@@ -63,32 +67,66 @@ export class Claims implements IClaims {
   }
 
   /**
-   * Verifies integrity of the claim, the claim is issued by the user
-   *  delegate and the authenticity of the issuer's signature
+   * Verifies issuance and publishing of claim at `claimUrl`
    *
    * @param claimUrl {string}
    * @param hashFns {{ [alg: string]: (data: string) => string }}
    */
   async verify(
-    claimUrl: string, hashFns?: { [alg: string]: (data: string) => string },
+    claimUrl: string,
+    {
+      hashFns,
+      issuerDoc,
+      holderDoc,
+    }: {
+      hashFns?: { [alg: string]: (data: string) => string };
+      issuerDoc?: IDIDDocument;
+      holderDoc?: IDIDDocument;
+    } = {}
   ): Promise<IPublicClaim | IPrivateClaim> {
-    const token = await this.store.get(claimUrl);
-    const claim = this.jwt.decode(token) as
-      (IPublicClaim | IPrivateClaim) & { iss: string; sub: string };
-    if (!(await this.verifySignature(token, claim.iss))) {
-      throw new Error('Invalid signature');
+    const token = await this.getServiceEndpointToken(claimUrl, {
+      hashFns,
+      holderDoc,
+    });
+    const claim = this.jwt.decode(token) as (IPublicClaim | IPrivateClaim) &
+      JwtPayload;
+    if (!issuerDoc) {
+      issuerDoc = await this.document.read(claim.iss);
     }
-    if (!this.document.isValidDelegate(DelegateTypes.verification, claim.signer, claim.did)) {
-      throw new Error('Issuer isn\'t a use\'r delegate');
+    const authVerifier = new AuthTokenVerifier(issuerDoc);
+    if (await authVerifier.verify(token)) {
+      return claim;
```

Comment:
I'm not sure we can necessarily rely on the `AuthTokenVerifier` exclusively in this way as we can't be sure that the token is an "auth" token. Because you can have token signed for "authentication" or "verification" (at least in our current code)

Like we have this code in `operator.ts`
```
/**
 * The two main types of public keys are, in turn, verification key and signature authentication key
 */
export enum PubKeyType {
  SignatureAuthentication2018 = 'sigAuth', VerificationKey2018 = 'veriKey'
}
```

Reply:
Not sure how approach this. Claim issuer should be authenticated, but the claim is not always authentication claim. Should we change `AuthTokenVerifier` to authenticate claims signed by verification key?